liquibase-core/src/main/java/liquibase/snapshot/jvm/ColumnSnapshotGenerator.java

```diff
@@ -404,6 +404,9 @@ protected DataType readDataType(CachedRow columnMetadataResultSet, Column column
             }
         } else if (database instanceof PostgresDatabase) {
             columnTypeName = database.unescapeDataTypeName(columnTypeName);
+            if (columnTypeName.startsWith("_")) {
+                columnTypeName = columnTypeName.replaceFirst("_", "").concat("[]");
+            }
```

Comment:
can you leave some comments on why we need to do this?

Reply:
@neupsh, I spent some time for debug snapshot of column logic, I tested it with DDL:
`CREATE TABLE table1 (
        id          integer PRIMARY KEY,
        table1_id   integer,
        short_arr int2[],
        int_arr int4[],
        long_arr int8[],
        float_arr float4[],
        double_arr float8[],
        boolean_arr bool[],
        string_arr varchar[],
        text_arr text[],
        integer_arr_v2 integer[]
      );`

And I saw that array comes as simple datatype, but with underscore before type name and liquibase produced smth like:
`# "bookshop"."table1" (
  #   "id" INT NOT NULL,
  #   "table1_id" INT,
  #   "short_arr" _INT2(5),
  #   "int_arr" _INT4(10),
  #   "long_arr" _INT8(19),
  #   "float_arr" _FLOAT4(8, 8),
  #   "double_arr" _FLOAT8(17, 17),
  #   "boolean_arr" _BOOL(1),
  #   "string_arr" _VARCHAR,
  #   "text_arr" _TEXT,
  #   "integer_arr_v2" _INT4(10)
  # )`
After debugging, I understood that remove underscore before type name and add square brackets after type name should works, but also we should remove default values, to achieve this, I added array with known data types to list of unmodifiable data types and it skips determination of default value.

Reply:
Thanks @achehrynets , I am not very well versed on this. I have requested a review from @nvoxland as well.

Reply:
My worry was that assuming a _ prefixed datatype means an array was too broad, but https://www.postgresql.org/message-id/20061016193942.GF23302%40svana.org says that internally array datatypes are defined with an underscore prefix. So it looks like we're doing the right thing here